services/plugin/src/Plugin.tsx

```diff
@@ -27,14 +27,23 @@ const getPluginEntryPoint = ({
 export const Plugin = ({
     pluginSource,
     pluginShortName,
-    ...propsToPass
+    ...propsToPassNonMemoized
 }: {
     pluginSource?: string
     pluginShortName?: string
     propsToPass: any
 }): JSX.Element => {
     const iframeRef = useRef<HTMLIFrameElement>(null)
 
+    // we do not know what is being sent in passed props, so for stable reference, memoize using JSON representation
+    const propsToPassNonMemoizedJSON = JSON.stringify(propsToPassNonMemoized)
```

Comment:
stable reference for props is handled in a more thought-out-manner in https://github.com/dhis2/app-runtime/pull/1351